src/managers/KeybindManager.cpp

```diff
@@ -1395,17 +1395,16 @@ SDispatchResult CKeybindManager::moveFocusTo(std::string args) {
         g_pCompositor->getWindowInDirection(PLASTWINDOW, arg);
 
     // Prioritize focus change within groups if the window is a part of it.
-    if (*PGROUPCYCLE) {
-        if (!PLASTWINDOW->m_sGroupData.pNextWindow.expired()) {
-            if (arg == 'l' && PLASTWINDOW != PLASTWINDOW->getGroupHead()) {
-                PLASTWINDOW->setGroupCurrent(PLASTWINDOW->getGroupPrevious());
-                return {};
-            }
+    if (*PGROUPCYCLE && (!PLASTWINDOW->m_sGroupData.pNextWindow.expired())) {
+        auto isTheOnlyGroupOnWs = (PWINDOWTOCHANGETO == nullptr) && (g_pCompositor->m_vMonitors.size() == 1);
```

Comment:
whats with all those parentheses. Also, `== nullptr` is the same as `!var`

Reply:
sorry, im new to this. i didn't see it. i'll fix it.

Reply:
done